lib/ProofOfPossessionBuilder.ts

```diff
@@ -0,0 +1,38 @@
+import { createProofOfPossession } from './functions';
+import { EndpointMetadata, PROOF_CANT_BE_CONSTRUCTED, ProofOfPossession, ProofOfPossessionArgs } from './types';
+
+export class ProofOfPossessionBuilder {
+  clientId?: string;
+  endpointMetadata: EndpointMetadata;
+  proof?: ProofOfPossession;
+  proofCallbackOpts?: ProofOfPossessionArgs;
+
+  withProofCallbackOpts(proofCallbackOpts: ProofOfPossessionArgs): ProofOfPossessionBuilder {
```

Comment:
Suggest to make this and the withProof static and rename them to fromX. So that it becomes clear you will either start from a proof or from the metatada